lib/ts/MemArena.cc

```diff
@@ -0,0 +1,160 @@
+/** @file
+
+    @section license License
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+#include "MemArena.h"
+#include <ts/ink_memory.h>
+#include <ts/ink_assert.h>
+
+using namespace ts;
+
+inline void
+MemArena::MemArenaBlock::alloc(size_t n)
+{
+  ink_assert(n > 0);
+
+  data      = ats_malloc(n);
+  size      = n;
+  allocated = 0;
+  next      = nullptr;
+}
+
+inline void
+MemArena::MemArenaBlock::free()
+{
+  if (data) {
```

Comment:
`free` works fine on a `null` pointer. Don't bother to check.

Reply:
ok. done.